Tests/CreateAPITests/GenerateOptionsTests.swift

```diff
@@ -657,4 +657,31 @@ final class GenerateOptionsTests: GenerateBaseTests {
         // THEN
         try compare(package: "petstore-identifiable")
     }
+    
+    func testPetstoreEntityExclude() throws {
+        // GIVEN
+        let command = try Generate.parse([
+            pathForSpec(named: "petstore", ext: "yaml"),
+            "--output", temp.url.path,
+            "--package", "petstore-entity-exclude",
+            "--generate", "entities",
+            "--config", config("""
+            entities:
+                exclude:
+                - Error
+                - Pet.id
+                - Store.pets
+            rename:
+                properties:
+                    Pet.id: notID
+                    Pet.name: id
```

Comment:
So if I understand this right, the logic has ended up excluding `notID` and included `id`. I guess that makes sense since its consistent with how entity naming templates work. 

For the sake of the test example, can you also please use `--entityname-template "%0DTO"` or something similar to confirm its working as expected when entities are renamed too? 🙇

Reply:
No, as the excluded filter happens first and then renaming happens in the next step when the property is made in `makeProperty`. So actually, the _original_ `Pet.id` was removed and then `Pet.name` is renamed to _id_. This was proven with iterative testing and would have been better shown with different property types (both are strings).

I personally think that these are the correct steps. `exclude` should only worry about excluding objects directly from the schema and the rename happens for entity generation. This is further backed up by the fact that we require renames to be in the format of the generated entity name.

Reply:
> please use `--entityname-template "%0DTO"` ... to confirm

Good catch as this does not work. The entity template is applied as jobs are created to generate the entity so the object type name is different than the exclude. However, I don't think that this should be correct. I think that the `entityname-template` should be applied _last_ as the entity is created for a similar reason as above for `exclude` and `rename`: use the schema object name for setting the excludes and renames. 

I don't think it is correct to specify:

```yaml
# using --entityname-template "$0Model.swift"

 entities:
   exclude:
   - PetModel.id
```
Additionally, I think `--entityname-template` [should be removed](https://github.com/CreateAPI/CreateAPI/issues/47#issuecomment-1205703517).